### PR TITLE
Update Proxy Server configuration

### PR DESCRIPTION
**Pull request by evan-m-jackson**

### CODE REVIEW

**Comment by paytonrules** [comment]
Commit messages here are much better - clearly state what's going on, make it much easier review the code as well (as I can see you did or didn't do what you meant). 

The next step is to make sure you've got a good summary and description in the PR. Many teams will have rules and guidelines for that, so it's good to develop that habit now.  Some example styles are here:

https://axolo.co/blog/p/part-3-github-pull-request-template

You don't need to use one of them but it's an example of what clients will be looking for when you get to that phase.

File: HTTPServerProject.Tests/ResponseTests.cs [hunk 1]
```diff
@@ -22,7 +22,7 @@ public void AddAndGetHeaderTest()
     {
         var expected = new List<string>() { "HTTP/1.1 200 OK", "Location: Earth", "", "Hello World!" };
         var writer = new TestWriteStreams(eStream);
-        var headers = new List<string>(){"Location: Earth"};
+        var headers = new List<string>() { "Location: Earth" };
```
Comment (paytonrules) [inline HTTPServerProject.Tests/ResponseTests.cs]: The commit here doesn't seem to match the actual code. Any idea what's going on here?

Comment (paytonrules) [reply HTTPServerProject.Tests/ResponseTests.cs]: Ah I see what happened - some commit message got re-used by mistake. This often happens in IntellJ because it has a (annoying IMO) feature of keeping the commit message from the last time you committed in the dialog. Make sure you double check commit messages when you commit.

Comment (evan-m-jackson) [reply HTTPServerProject.Tests/ResponseTests.cs]: Yes, my mistake. That was one of my first commits using Rider and I double tapped the commit message.

File: HTTPServerProject.Tests/HTTPServerProxyTests/TestProxyClient.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
+using HTTPServerProject.Proxy.Client;
+
+namespace HTTPServerProject.Tests.Proxy.Client;
+
+public class TestProxyClient : IProxyClient
```
Comment (paytonrules) [inline HTTPServerProject.Tests/HTTPServerProxyTests/TestProxyClient.cs]: It's not in this change - but what's the point of the empty interface `IProxyClient`?

File: HTTPServerProject.Tests/HTTPServerReadTests/BodyTests.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+{
+    public List<string> request = new List<string>() { "GET / HTTP/1.1", "Host: localhost:5000", "User-Agent: curl/7.79.1", "Accept: */*", "", "quit" };
+    
+    [Fact]
+    public void GetRequestBodyTest()
+    {
+        var reader = new TestReadStreams(request);
+
+        var header = new Header(reader);
+        var initialLine = header.GetLine();
```
Comment (paytonrules) [inline HTTPServerProject.Tests/HTTPServerReadTests/BodyTests.cs]: I don't think you need any of the header stuff in this test.

File: HTTPServerProject.Tests/HTTPServerReadTests/BodyTests.cs [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        var headers = header.GetHeaders();
+
+        var body = new Body(reader);
+        var expected = body.GetBody();
```
Comment (paytonrules) [inline HTTPServerProject.Tests/HTTPServerReadTests/BodyTests.cs]: This is `actual` not `expected` - your expected value is the hard coded "quit".

**Review by paytonrules — CHANGES_REQUESTED** [review]
The review isn't done - as I mentioned in chat I ran out of time this morning - but I had a few comments I wanted you to see right away.

File: HTTPServerProject.Tests/HTTPServerReadTests/BodyTests.cs (hunk 3 above)
Comment (evan-m-jackson) [reply HTTPServerProject.Tests/HTTPServerReadTests/BodyTests.cs]: Agreed.

File: HTTPServerProject.Tests/HTTPServerReadTests/BodyTests.cs (hunk 4 above)
Comment (evan-m-jackson) [reply HTTPServerProject.Tests/HTTPServerReadTests/BodyTests.cs]: Agreed.

File: HTTPServerProject.Tests/HTTPServerProxyTests/TestProxyClient.cs (hunk 2 above)
Comment (evan-m-jackson) [reply HTTPServerProject.Tests/HTTPServerProxyTests/TestProxyClient.cs]: It was just a way to make a test double for the `TcpClient` class, which I wrapped in the `ProxyClient` class

Comment (paytonrules) [reply HTTPServerProject.Tests/HTTPServerProxyTests/TestProxyClient.cs]: But what's being doubled? There's no methods on IProxyClient so there's nothing to double.

File: HTTPServerProject.Tests/HTTPServerResponseTests/StatusCodeTests.cs [hunk 5]
```diff
@@ -0,0 +1,19 @@
+using HTTPServerProject.Responses.Code;
+
+namespace HTTPServerProject.Tests.StatusCode;
+
+public class UnitTestsForWritingResponses
+{
+    [Theory]
+    [InlineData(200, "HTTP/1.1 200 OK")]
```
Comment (paytonrules) [inline HTTPServerProject.Tests/HTTPServerResponseTests/StatusCodeTests.cs:8]: Neat!

File: HTTPServerProject/HTTPServerResponse/ResponseByPath.cs [hunk 6]
```diff
@@ -7,8 +7,8 @@
 using HTTPServerProject.Responses.Write;
 using HTTPServerProject.WriteStream;
 
-namespace HTTPServerProject.Path
-{
+namespace HTTPServerProject.Path;
+
```
Comment (paytonrules) [inline HTTPServerProject/HTTPServerResponse/ResponseByPath.cs]: It's not part of this PR - but I noticed it looking at this one. RequestPath doesn't need a default and probably shouldn't use one. You're always injecting a dictionary - which is a good thing! - so you probably don't want the user to be able to create a RequestPath without a dictionary. It would be empty, and essentially useless.

File: HTTPServerProject.Tests/HTTPServerProxyTests/TestProxyClient.cs (hunk 2 above)
Comment (evan-m-jackson) [reply HTTPServerProject.Tests/HTTPServerProxyTests/TestProxyClient.cs]: I would have put the `GetStream()` method in IProxyClient but the `ProxyClient` returns a `NetworkStream` and the `TestProxyClient` returns a `List<string>`.  Wouldn't that cause an error in the interface?

File: HTTPServerProject/HTTPServerResponse/ResponseByPath.cs (hunk 6 above)
Comment (evan-m-jackson) [reply HTTPServerProject/HTTPServerResponse/ResponseByPath.cs]: Agreed.

File: HTTPServerProject.Tests/HTTPServerResponseTests/StatusCodeTests.cs (hunk 5 above)
Comment (evan-m-jackson) [reply HTTPServerProject.Tests/HTTPServerResponseTests/StatusCodeTests.cs:8]: Thanks 🤙🏽

**Comment by paytonrules** [comment]
With 48 commits you're gonna want so squash these into a clearer history. Probably not just one, but a few that explain what you did. (For instance you have two commits that say "Light Refactor of Code" - that could be one commit with a better title of what you did to refactor the code.)

**Comment by paytonrules** [comment]
There's a ton here now - let's get a better title and description on the PR.

File: HTTPServerProject.Tests/HTTPServerWriteTests/TestWriteStreams.cs [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+            {
+                sArr.Add("");
+            }
+            else
+            {
+                sArr.Add(str);
+            }
+        }
+
+        public void Flush()
```
Comment (paytonrules) [inline HTTPServerProject.Tests/HTTPServerWriteTests/TestWriteStreams.cs:38]: I'm surprised these don't have booleans set in them - you could use these to test if something is flushed and closed.

File: HTTPServerProject.Tests/HTTPServerProxyTests/TestClient.cs [hunk 8]
```diff
@@ -0,0 +1,16 @@
+using HTTPServerProxy.Client;
+
+namespace HTTPServerProxyTests.Client;
+
+public class TestProxyClient : IProxyClient
+{
+    List<string> stream = new List<string>();
+    public TestProxyClient()
```
Comment (paytonrules) [inline HTTPServerProject.Tests/HTTPServerProxyTests/TestClient.cs:8]: I don't believe the empty constructor is necessary.

File: HTTPServerProject.Tests/HTTPServerProxyTests/TestClient.cs [hunk 9]
```diff
@@ -0,0 +1,16 @@
+using HTTPServerProxy.Client;
+
+namespace HTTPServerProxyTests.Client;
+
+public class TestProxyClient : IProxyClient
```
Comment (paytonrules) [inline HTTPServerProject.Tests/HTTPServerProxyTests/TestClient.cs:5]: See my broader note attached to the bottom about `IProxyClient`

File: HTTPServerProject.Tests/HTTPServerProxyTests/TestFirstID.cs [hunk 10]
```diff
@@ -0,0 +1,103 @@
+using HTTPServerRead.Streams;
+using HTTPServerRead.Header;
+using HTTPServerRead.Body;
+using HTTPServerWrite.Request;
+using HTTPServerWrite.Streams;
+using HTTPServerProxy.FirstID;
+
+namespace HTTPServerProxyTests.FirstID;
+
+public class TestGetFirstID : IGetFirstID
```
Comment (paytonrules) [inline HTTPServerProject.Tests/HTTPServerProxyTests/TestFirstID.cs:10]: What purpose does this serve? It looks like a complete copy of `GetFirstID`.

File: HTTPServerProject.Tests/HTTPServerWriteTests/TestWriteStreams.cs (hunk 7 above)
Comment (evan-m-jackson) [reply HTTPServerProject.Tests/HTTPServerWriteTests/TestWriteStreams.cs:38]: Agreed.

File: HTTPServerProject.Tests/HTTPServerProxyTests/TestFirstID.cs (hunk 10 above)
Comment (evan-m-jackson) [reply HTTPServerProject.Tests/HTTPServerProxyTests/TestFirstID.cs:10]: I  changed the `WriteToDoListRequest` method in the `TestGetFirstID` to make a test double.

File: HTTPServerProject.Tests/HTTPServerProxyTests/TestClient.cs (hunk 9 above)
Comment (evan-m-jackson) [reply HTTPServerProject.Tests/HTTPServerProxyTests/TestClient.cs:5]: I am going to remove the empty interface `IProxyClient` and this unneccessary test double

File: HTTPServerProject.Tests/HTTPServerProxyTests/TestClient.cs (hunk 8 above)
Comment (evan-m-jackson) [reply HTTPServerProject.Tests/HTTPServerProxyTests/TestClient.cs:8]: Agreed.
